Source/Dafny/Verifier/ProofObligationDescription.cs

```diff
@@ -244,7 +244,7 @@ public class PreconditionSatisfied : ProofObligationDescription {
       : $"error is impossible: {customErrMsg}";
 
   public override string FailureDescription =>
-    customErrMsg ?? "possible violation of function precondition";
+    customErrMsg ?? "function precondition might not hold";
```

Comment:
Wonderful ! That has resulted in a lot of test fixes (good job!) but this message is much better.